### DISCUSSION TITLE
Encode and Write stages look similar

### DISCUSSION THREAD

**Discussion by jotak** [netobserv/flowlogs-pipeline#224] [Ideas]
**Encoder**

```go
type Encoder interface {
	Encode(in []config.GenericMap)
}
```

**Writer**

```go
type Writer interface {
	Write(in []config.GenericMap)
}
```

Shouldn't they be the same?
Or, actually, shouldn't `Encoder` return an `[]interface{}`, mirroring `Decoder` ? So that, for instance, we could have a JSON encoder? And the current Prometheus and Kafka encoders would be moved to writers?

**Comment by KalmanMeth** [comment]
Originally Encoder was not a final stage, and fed into Write (using `[]interface{}`). We changed this to make Encoder a final stage, similar to Write, making encode_prom a final stage. They could now be merged into a single type of stage (called Export or Final or Terminal or whatever).
